buildSrc/src/main/java/org/elasticsearch/gradle/NoticeTask.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.gradle;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.OutputFile;
+import org.gradle.api.tasks.TaskAction;
+
+
+/**
+ * A task to create a notice file which includes dependencies' notices.
+ */
+public class NoticeTask extends DefaultTask {
+
+    private static final Charset UTF8 = Charset.forName("UTF-8");
+    @InputFile
+    private File inputFile = getProject().getRootProject().file("NOTICE.txt");
+
+    @OutputFile
+    private File outputFile = new File(getProject().getBuildDir(), "notices/" + getName() + "/NOTICE.txt");
+
+    /**
+     * Directories to include notices from
+     */
+    private List<File> licensesDirs = new ArrayList<>();
+
+    public NoticeTask() {
+        setDescription("Create a notice file from dependencies");
+        // Default licenses directory is ${projectDir}/licenses (if it exists)
+        File licensesDir = new File(getProject().getProjectDir(), "licenses");
+        if (licensesDir.exists()) {
+            licensesDirs.add(licensesDir);
+        }
+    }
+
+    /**
+     * Add notices from the specified directory.
+     */
+    public void setLicensesDir(File licensesDir) {
```

Comment:
The method was not named as a setter in groovy so this could be DSL-like. ie, usage looks like (notice the lack of equals sign):

```
noticeTask {
  licensesDir 'foo'
}
```

Reply:
Note that this is different than setting a single property as it adds the inputs to the list.